### DISCUSSION TITLE
Smart way to update

### DISCUSSION THREAD

**Discussion by swa72** [emsesp/EMS-ESP32#1968] [General]
I read https://emsesp.org/Getting-Started/ and still ask myself, what is a smooth way to update (without messing up HA too much) ? In my case from 3.5.1 to 3.6.5.
* stop mqtt on 3.5.1 
* remove all topics using MQTT explorer
* update to 3.6.5 
* clean up entities in HA afterwards

Any hints?

**Comment by proddy** [comment]
I would jump to 3.7.0 dev which is close to be released, then in MQTT settings set to multiple instances and pick a new name for the MQTT base. Then you'll have the new entities which you can check in a new HA dashboard. If that works out fine, then follow your steps above.

**Comment by swa72** [comment]
Thanks for the quick reply. https://emsesp.org/#new-features-introduced-in-36 reads _"When upgrading to v3.6 for the first time from v3.4 on a BBQKees Gateway board you will need to use the [EMS-EPS Flasher](https://github.com/emsesp/EMS-ESP-Flasher/releases) to correctly"._

Do I have to use the flasher as well (coming from 3.5)? 

I tried a bunch of images and got either an "Internal Server Error" or, even funnier, the device reported a successful upgrade. I rebooted just to find the same version number again (3.5.1).

**Reply by proddy** [reply]
There have been many changes since 3.5.1 was release 1.5 years ago, including the partition structure. If it's not too much hassle for you, i would make a copy of the settings and any customizations and then erase the ESP32 and upload the latest dev-33 which will be released today or tomorrow. 

Are you still on the older ESP32's with 4MB and no PSRAM?

**Comment by swa72** [comment]
To be honest, I dunno. I've identified the device as one of BBQKees gateway s32 - 2021 (https://bbqkees-electronics.nl/wiki/gateway/connecting-a-gateway-s32-or-s3.html)

**Reply by proddy** [reply]
thats an ESP32 4MB then. I'll send a link to the firmware when its ready
